### PR TITLE
Initial BTF support

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
This is currently being used by "replace" in my repository at:

https://github.com/rafaeldtinoco/tracee/tree/external-btf

That repository needs:

replace github.com/aquasecurity/libbpfgo v0.1.1 => ./3rdparty/libbpfg

in go.mod, together with these changes, in order to work.

### CODE REVIEW

File: libbpfgo.go [hunk 1]
```diff
@@ -297,33 +297,68 @@ func errptrError(ptr unsafe.Pointer, format string, args ...interface{}) error {
...
+	C.free(unsafe.Pointer(bpfFile))
+	C.free(unsafe.Pointer(btfFile))
+
 	return &Module{
 		obj: obj,
 	}, nil
 }
 
-func NewModuleFromBuffer(bpfObjBuff []byte, bpfObjName string) (*Module, error) {
+func NewModuleFromBuffer(btfObjPath string, bpfObjBuff []byte, bpfObjName string) (*Module, error) {
```
Comment (grantseltzer) [inline libbpfgo.go:331]: Don't think we need to change the function signature here if not using `btfObjPath`

File: libbpfgo.go [hunk 2]
```diff
@@ -297,33 +297,68 @@ func errptrError(ptr unsafe.Pointer, format string, args ...interface{}) error {
 }
 
 func NewModuleFromFile(bpfObjFile string) (*Module, error) {
+	return NewModuleFromFileBtf("", bpfObjFile)
+}
+
+func NewModuleFromFileBtf(btfObjPath string, bpfObjFile string) (*Module, error) {
```
Comment (grantseltzer) [inline libbpfgo.go:303]: For consistencies sake can you change bpfObjFile -> bpfObjPath.

**Review by grantseltzer — CHANGES_REQUESTED** [review]

**Comment by grantseltzer** [comment]
Is the libbpf sub module upgrade needed? The specific commit this changes to is a change to the libbpf mirror CI

File: libbpfgo.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package libbpfgo
 
 /*
-#cgo LDFLAGS: -lelf -lz
+#cgo LDFLAGS: -lelf -lz -lbpf
```
Comment (yanivagman) [inline libbpfgo.go:4]: For tracee, we use our own libbpf submodule (a statically compiled .a), so adding -lbpf will not work with it

Comment (rafaeldtinoco) [reply libbpfgo.go:4]: this was an error for my last push and local checks, wasn't meant to be there. thank you

**Review by grantseltzer — DISMISSED** [review]
LGTM! :shipit:

**Comment by rafaeldtinoco** [comment]
I'll need another +1 as i had to have the accidentally added -lbpf LDFLAG during my last push. Thanks and sorry!

**Review by grantseltzer — APPROVED** [review]

File: libbpfgo.go [hunk 4]
```diff
@@ -296,34 +296,69 @@ func errptrError(ptr unsafe.Pointer, format string, args ...interface{}) error {
 	return fmt.Errorf(format+": %v", args...)
 }
 
-func NewModuleFromFile(bpfObjFile string) (*Module, error) {
+func NewModuleFromFile(bpfObjPath string) (*Module, error) {
+	return NewModuleFromFileBtf("", bpfObjPath)
+}
+
+func NewModuleFromFileBtf(btfObjPath string, bpfObjPath string) (*Module, error) {
```
Comment (itaysk) [inline libbpfgo.go:303]: should be `NewModuleFromFileBTF` (inconsistent with https://github.com/aquasecurity/libbpfgo/pull/53/ and with golang naming convensions)

Comment (rafaeldtinoco) [reply libbpfgo.go:303]: https://github.com/aquasecurity/libbpfgo/pull/54

**Pull request merged**